comms/dht/src/actor.rs

```diff
@@ -441,14 +438,32 @@ impl<'a> DhtActor<'a> {
 
                 Ok(candidates)
             },
+            Propagate(mut exclude) => {
+                let mut candidates = Self::select_closest_peers_for_propagation(
+                    &config,
+                    &peer_manager,
+                    node_identity.node_id(),
+                    config.num_neighbouring_nodes,
+                    &exclude,
+                    PeerFeatures::MESSAGE_PROPAGATION,
+                )
+                .await?;
+
+                exclude.extend(candidates.iter().map(|p| p.node_id.clone()));
+                let mut random_peers = peer_manager
+                    .random_peers(config.num_random_propagation_nodes, &exclude)
```

Comment:
You select a new set of random peers for each message. Intuitively I think a random set _that worked well last time_ is a more robust approach (though I have no evidence to back that up).

Reply:
Working on a better solution inside comms where it's more appropriate to keep track of connections. It'll take a few days. But you are correct, this approach doesn't scale.

Reply:
More specifically (and the reason this is just a test draft) it will eventually open connections to every base node it can.

Reply:
:+1: